Formula/ola.rb

```diff
@@ -61,9 +50,12 @@ def install
     system "autoreconf", "--force", "--install", "--verbose"
     system "./configure", *std_configure_args, *args
     system "make", "install"
+    system "make", "check", "VERBOSE=1"
   end
 
   test do
+    system bin/"olad", "-h"
+    system bin/"olad", "--help"
```

Comment:
> We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In this specific case it would catch the particular issue we've ended up with which is that options parsing isn't working (seemingly only on Homebrew, as building our upcoming release with minimal changes via GitHub Action's Macs is fine, and likewise on users machines.

>     * If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.

We've got that test for the Python library already which someone added, and there's a similar one for our C library in Debian we should be able to port across too.

> In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

>     * If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.

The issue with this particular tool, and indeed arguably with the whole suite is that it all interoperates. olad is the server, and the other tools are all clients to interrogate it. A handful have some sort of config validation option, but most are either designed to query a particular piece of information from the server, or to send/receive data with it. Obviously the server is intended to run all the time, so you'd need to:

1. Start the server
2. Send some data
3. Receive some data
4. Validate the exchange
5. Stop the server

Instead we primarily concentrate on high quality unit tests, which indeed have found the issue I'm trying to track down, only they aren't run so we've had to wait for a user's error report to find out about them.

Reply:
I'd also welcome advice for the case where we do have a good operational test of one of our program where it confirms it can validate some test files, that's already in our unit tests here:
https://github.com/OpenLightingProject/ola/blob/49cf6567d1be0b82508f3a10cd91b65286d30d5b/tools/ola_trigger/Makefile.mk#L77-L79

Is there a better way to add it to Homebrew than having to manually duplicate it, and the test data, into your config file and the risk of mistakes, or particularly missing adding new test cases?